### PR TITLE
[Merged by Bors] - chore: remove unnecessary NeBot assumptions

### PR DESCRIPTION
**Pull request by D-Thomine**
I analysed all the use of NeBot assumptions (that a filter needs to be non-empty). I found 10 lemmas where this assumption was unnecessary, and removed it (with some modification of the proofs when necessary). When possible, I simplified the proofs which use those lemmas.

Unrelated: I cleaned up some lemmas I encountered which had universal quantifiers/chains of implications in their conclusions, using additional assumptions instead.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Calculus/UniformLimitsDeriv.lean [hunk 1]
```diff
@@ -419,20 +416,19 @@ _uniformly_ to their limit on an open set containing `x`. -/
 theorem hasFDerivAt_of_tendstoUniformlyOn [NeBot l] {s : Set E} (hs : IsOpen s)
     (hf' : TendstoUniformlyOn f' g' l s)
     (hf : ∀ n : ι, ∀ x : E, x ∈ s → HasFDerivAt (f n) (f' n x) x)
-    (hfg : ∀ x : E, x ∈ s → Tendsto (fun n => f n x) l (𝓝 (g x))) :
-    ∀ x : E, x ∈ s → HasFDerivAt g (g' x) x := fun _ =>
-  hasFDerivAt_of_tendstoLocallyUniformlyOn hs hf'.tendstoLocallyUniformlyOn hf hfg
+    (hfg : ∀ x : E, x ∈ s → Tendsto (fun n => f n x) l (𝓝 (g x))) (hx : x ∈ s) :
+    HasFDerivAt g (g' x) x := hasFDerivAt_of_tendstoLocallyUniformlyOn
+      hs hf'.tendstoLocallyUniformlyOn hf hfg hx
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/Calculus/UniformLimitsDeriv.lean:421]: Splitting the line there is not very readable, IMO. I think this is better:
```suggestion
    HasFDerivAt g (g' x) x :=
  hasFDerivAt_of_tendstoLocallyUniformlyOn hs hf'.tendstoLocallyUniformlyOn hf hfg hx
```

File: Mathlib/Analysis/Calculus/UniformLimitsDeriv.lean [hunk 2]
```diff
@@ -541,18 +537,17 @@ theorem hasDerivAt_of_tendsto_locally_uniformly_on' [NeBot l] {s : Set 𝕜} (hs
 theorem hasDerivAt_of_tendstoUniformlyOn [NeBot l] {s : Set 𝕜} (hs : IsOpen s)
     (hf' : TendstoUniformlyOn f' g' l s)
     (hf : ∀ᶠ n in l, ∀ x : 𝕜, x ∈ s → HasDerivAt (f n) (f' n x) x)
-    (hfg : ∀ x : 𝕜, x ∈ s → Tendsto (fun n => f n x) l (𝓝 (g x))) :
-    ∀ x : 𝕜, x ∈ s → HasDerivAt g (g' x) x := fun _ =>
-  hasDerivAt_of_tendstoLocallyUniformlyOn hs hf'.tendstoLocallyUniformlyOn hf hfg
+    (hfg : ∀ x : 𝕜, x ∈ s → Tendsto (fun n => f n x) l (𝓝 (g x))) (hx : x ∈ s) :
+    HasDerivAt g (g' x) x := hasDerivAt_of_tendstoLocallyUniformlyOn
+      hs hf'.tendstoLocallyUniformlyOn hf hfg hx
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/Calculus/UniformLimitsDeriv.lean:542]:
```suggestion
    HasDerivAt g (g' x) x :=
  hasDerivAt_of_tendstoLocallyUniformlyOn hs hf'.tendstoLocallyUniformlyOn hf hfg hx
```

**Review by Ruben-VandeVelde — APPROVED** [review]
Thanks! Looks reasonable overall.

File: Mathlib/Analysis/Calculus/UniformLimitsDeriv.lean (hunk 1 above)
Comment (D-Thomine) [reply Mathlib/Analysis/Calculus/UniformLimitsDeriv.lean]: Done, thanks!

File: Mathlib/Analysis/Calculus/UniformLimitsDeriv.lean (hunk 2 above)
Comment (D-Thomine) [reply Mathlib/Analysis/Calculus/UniformLimitsDeriv.lean]: Done.

**Review by Ruben-VandeVelde — APPROVED** [review]
Thanks!

maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**